core/src/main/kotlin/event/channel/thread/ThreadDeleteEvent.kt

```diff
@@ -0,0 +1,42 @@
+package dev.kord.core.event.channel.thread
+
+import dev.kord.core.entity.channel.Channel
+import dev.kord.core.entity.channel.thread.DeletedThreadChannel
+import dev.kord.core.entity.channel.thread.NewsChannelThread
+import dev.kord.core.entity.channel.thread.TextChannelThread
+import dev.kord.core.entity.channel.thread.ThreadChannel
+import dev.kord.core.event.channel.ChannelCreateEvent
+import dev.kord.core.event.channel.ChannelDeleteEvent
+
+interface ThreadChannelDeleteEvent : ChannelDeleteEvent {
+    override val channel: DeletedThreadChannel
+}
```

Comment:
This can be sealed

```suggestion
sealed interface ThreadChannelDeleteEvent : ChannelDeleteEvent {
    override val channel: DeletedThreadChannel
}
```